Src/library/base/elks/kernel/routine.e

```diff
@@ -135,16 +134,43 @@ feature -- Status report
 			elseif args.count >= open_count then
 				from
 					Result := True
-					open_type_codes := eif_gen_typecode_str ($Current)
 					i := 1
 				until
 					i > open_count or not Result
 				loop
 					arg_type_code := args.item_code (i)
-					Result := arg_type_code = open_type_codes.item (i).code
-					if Result and then arg_type_code = {TUPLE}.reference_code then
+					l_type := open_operand_type (i)
+					inspect arg_type_code
+					when {TUPLE}.boolean_code then
+						Result := l_type = ({BOOLEAN}).type_id
+					when {TUPLE}.character_8_code then
+						Result := l_type = ({CHARACTER_8}).type_id
+					when {TUPLE}.character_32_code then
+						Result := l_type = ({CHARACTER_32}).type_id
+					when {TUPLE}.integer_8_code then
+						Result := l_type = ({INTEGER_8}).type_id
+					when {TUPLE}.integer_16_code then
+						Result := l_type = ({INTEGER_16}).type_id
+					when {TUPLE}.integer_32_code then
+						Result := l_type = ({INTEGER_32}).type_id
+					when {TUPLE}.integer_64_code then
+						Result := l_type = ({INTEGER_64}).type_id
+					when {TUPLE}.natural_8_code then
+						Result := l_type = ({NATURAL_8}).type_id
+					when {TUPLE}.natural_16_code then
+						Result := l_type = ({NATURAL_16}).type_id
+					when {TUPLE}.natural_32_code then
+						Result := l_type = ({NATURAL_32}).type_id
+					when {TUPLE}.natural_64_code then
+						Result := l_type = ({NATURAL_64}).type_id
+					when {TUPLE}.pointer_code then
+						Result := l_type = ({POINTER}).type_id
+					when {TUPLE}.real_32_code then
+						Result := l_type = ({REAL_32}).type_id
+					when {TUPLE}.real_64_code then
+						Result := l_type = ({REAL_64}).type_id
+					when {TUPLE}.reference_code then
```

Comment:
Believe it or not, this new implementation is on average faster than the old one. And it avoids the call to the ISE's specific external routine 'eif_gen_typecode_str'.
My benchmarks, calling 10,000,000 times 'valid_operands'.
Argument containing only references: old 2.7 seconds, new 1.5 seconds
Argument containing more basic types than references: old: 2.2 seconds, new: 1.4 seconds
Now, resetting 'open_types' to Void before each call:
Argument containing only references: old 6.3 seconds, new 5.2 seconds
Argument containing more basic types than references: old: 3.7 seconds, new: 5.3 seconds
This last one is the only case where the old implementation is faster.